featurebyte/service/feature_table_cache.py

```diff
@@ -1,111 +1,367 @@
+                prepared_node_name,
+            ) = await self.namespace_handler.prepare_graph_to_store(
+                graph=pruned_graph,
+                node=pruned_graph.get_node_by_name(node_name_map[node.name]),
+                sanitize_for_definition=True,
+            )
+            definition_hash_extractor = DefinitionHashExtractor(graph=prepared_graph)
+            definition_hash = definition_hash_extractor.extract(
+                prepared_graph.get_node_by_name(prepared_node_name)
+            ).definition_hash
+            hashes[definition_hash] = node
+
+        cached_hashes = set(
+            feat.definition_hash for feat in feature_table_cache_metadata.feature_definitions
+        )
+        cached = {
+            definition_hash: node
+            for definition_hash, node in hashes.items()
+            if definition_hash not in cached_hashes
+        }
+        non_cached = {
+            definition_hash: node
+            for definition, node in hashes.items()
+            if definition_hash in cached_hashes
+        }
+        return cached, non_cached
+
+    async def _create_table(
+        self,
+        feature_store: FeatureStoreModel,
+        observation_table: ObservationTableModel,
+        db_session: BaseSession,
+        final_table_name: str,
+        graph: QueryGraph,
+        nodes: Dict[str, Node],
+        serving_names_mapping: Optional[Dict[str, str]] = None,
+        is_feature_list_deployed: bool = False,
+    ) -> None:
+        intermediate_table_name = (
+            f"__TEMP__{MaterializedTableNamePrefix.FEATURE_TABLE_CACHE}_{ObjectId()}"
+        )
+        request_column_names = {col.name for col in observation_table.columns_info}
+        parent_serving_preparation = (
+            await self.entity_validation_service.validate_entities_or_prepare_for_parent_serving(
+                graph=graph,
+                nodes=list(nodes.values()),
+                request_column_names=request_column_names,
+                feature_store=feature_store,
+                serving_names_mapping=serving_names_mapping,
+            )
+        )
+        output_table_details = TableDetails(
+            database_name=db_session.database_name,
+            schema_name=db_session.schema_name,
+            table_name=intermediate_table_name,
+        )
+        await get_historical_features(
+            session=db_session,
+            tile_cache_service=self.tile_cache_service,
+            graph=graph,
+            nodes=list(nodes.values()),
+            observation_set=observation_table,
+            feature_store=feature_store,
+            output_table_details=output_table_details,
+            serving_names_mapping=serving_names_mapping,
+            is_feature_list_deployed=is_feature_list_deployed,
+            parent_serving_preparation=parent_serving_preparation,
+        )
+
+        feature_names = [
+            expressions.alias_(
+                node.name,
```

Comment:
This should use `get_node_output_column_name` to obtain the feature name for each node. `node.name` is the name of the node, not the feature represented by the node.